### ISSUE TITLE
About attention map visualization and mimicking the label

### ISSUE THREAD

**Issue by psychotechTD** [cardwing/Codes-for-Lane-Detection#264]
Hi cardwing, thanks for your sharing.
After reading your paper and code in  Enet-Label-torch, I am confused about the following two questions:

1. I can't get similar attention maps as shown in the paper
![image](https://user-images.githubusercontent.com/13014370/78172682-103d0f00-7489-11ea-9a3a-1537fd964000.png)
As referred in #131 
attention maps are obtained via sequentially apllying pow(), sum() and softmax on the _hxw_ dimension, so that the _hxw_ elements in attention map add up to one. but in practice, the values after softmax are very polarized, most of the values are close to zero and only few pixels can reach up to 0.5 and above. 
But for the map shown in the paper, the sum of pixel values in attetnion maps obviously is bigger than 1,  do you apply a normalization on softmax result?
And the values in your attention maps are smoothly distibuted, many of them are around 0.5(the green area),why? (in my test, only very few of the point are close to 1 and the left are close to zeros, which is more like a softmax result )
According to the image above, the values represent probablity of lane existence, how come the pixel values in attention maps shows the probability of lane existence? 
My attention maps after doing pow, sum, (spatialsoftmax) and normalization on the block4 output are like this. None of them shows the shape of lanes well(the first one without spatialsoftmax and second one with saptialsoftmax)
![no softmax(just pow() and sum() then normalized)](https://user-images.githubusercontent.com/13014370/78179040-26e86380-7493-11ea-86b1-d927e0fc5902.png)
![with softmax(pow(), sum(),softmax(),normalized)](https://user-images.githubusercontent.com/13014370/78179230-72027680-7493-11ea-806c-ff64d477657b.png)
Could you please point out where I understand wrong?

2. About attention output mimicking the lable
As described in #208 , according to the code below in train.lua and init.lua. target[1] is the segmentation label, target[2] is the existence label. 
```lua
local tmp_target = [{}]
table.insert(tmp_target, self.target[1])
table.insert(tmp_target, self.target[2])
table.insert(tmp_target, self.target[1])
table.insert(tmp_target, self.target[1]) 
self.target = tmp_target
```
```lua
local SCE = cudnn.SpatialCrossEntropyCriterion(weights):cuda() 
local BCE = nn.BCECriterion():cuda() 
local MSE_1 = nn.MSECriterion():cuda() 
local MSE_2 = nn.MSECriterion():cuda() 
criterion = nn.ParallelCriterion2():add(SCE, 1):add(BCE, 0.1):add(MSE_1, 0.0):add(MSE_2, 0.0) -- set the coefficients of MSE_1 and MSE_2 to be 0.1 if you want to use the distillation loss 
```
that seems to indicate SCE(segmentation_map,seg_label), BCE(existence, exist_label),MSE(block3,seg_label),MSE(block4,seg_label), which is the attention output mimicking the labels.
But the value in attention output of block3 and block4 are between 0 and 1. values of  segmentation label are like 1,2,3,4,5. How does the mimicking work?

**Comment by cardwing** [comment]
1. Sorry about the unclarity. The attention maps used to visualize are obtained via performing pow and sum (without spatialsoftmax). The values in the attention map do not actually represent probability of lane existence (It only indicates where the network is looking or focusing). Besides, do not crop the original image when performing visualization (current code crops the original image to obtain more gains).
2. This mimicking strategy is similar to deep supervision (Deeply-Supervised Nets). You can have a try and check its performance gains. Several engineering tricks are mentioned in other issues. You can also have a thorough read first.

**Comment by psychotechTD** [comment]
thanks for your reply, I will try your advise
